sql/mysqld.cc

```diff
@@ -5587,6 +5589,71 @@ static void test_lc_time_sz()
 }
 #endif//DBUG_OFF
 
+static void check_upgrade(void)
+{
+  File info_file;
+  char upgrade_from_version[50]= {0};
+  size_t length;
+  ulong version_id;
+
+  uint major, minor, version;
+  char *point= upgrade_from_version, *end_point, *pos;
+
+  DBUG_ENTER("check_upgrade");
+
+  if (upgrade_info_file != 0)
+  {
+    fn_format(upgrade_info_file, "mysql_upgrade_info", mysql_real_data_home_ptr, "", MYF(0));
```

Comment:
Need to check for `mariadb_upgrade_info` first, then `mysql_upgrade_info` if that is not found.

Reply:
I've been thinking about this, as `mariadb_upgrade_info` and `mysql_upgrade_info` files are not going to co-exists in any versions , do you think is it worth it to add extra logic for checking both (it will be required not only here but also on the MTR tests)?

Versions `<11.0` will always look for `mysql_upgrade_info` and versions `>=11.0` will always look for `mariadb_upgrade_info`, so I think we can just merge this on `10.11` using `mysql_upgrade_info` and just rename the file again when merging the code into `11.0`. Probably it would have been good to had this merged before https://github.com/MariaDB/server/pull/2475 so we could have just address that there.

Reply:
As this is a new feature, it won't be going into 10.11, it would only go in 11.0 (or newer depending on when merged). >=11.0 for now needs to check for both in case it is the 10.11 packages are replaced with 11.0.

Reply:
For that case, what do you think about just looking for `mariadb-upgrade-status`? A message will also be logged if the file doesn't exists, so this way we keep it simple and we don't have to worry about stop looking for the file in the future versions.

Reply:
I have renamed the file to use only `mariadb_upgrade_info` as mentioned in my previous comment and updated the PR to target 11.0 branch instead. Please let me know if you thing I should add/change anything else.

Reply:
It is conceivable that `mysql_upgrade_info` will exist from a pre-11.0 version when someone upgrades to 11.0. So I think both checks need to exist.

Reply:
Yep, I meant that with the current changes we're only looking for `mariadb_upgrade_info` and in case it doesn't exist this will be logged, so I think we wouldn't need to worry about `mysql_upgrade_info` :

https://github.com/MariaDB/server/blob/df9bf90bb34d577351dcc2ca66e161c83575c719/sql/mysqld.cc#L5641

Let me know if it's clearer now, or if you just prefer to check for  `mysql_upgrade_info` and I will add the changes.